splink/blocking_rule_library.py

```diff
@@ -0,0 +1,169 @@
+from typing import Union, final
+
+from sqlglot import TokenError, parse_one
+
+from .blocking_rule_creator import BlockingRuleCreator
+from .column_expression import ColumnExpression
+from .dialects import SplinkDialect
+
+
+def _translate_sql_string(
+    sqlglot_base_dialect_sql: str,
+    to_sqlglot_dialect: str,
+    from_sqlglot_dialect: str = None,
+) -> str:
+    tree = parse_one(sqlglot_base_dialect_sql, read=from_sqlglot_dialect)
+
+    return tree.sql(dialect=to_sqlglot_dialect)
+
+
+class ExactMatchRule(BlockingRuleCreator):
+    def __init__(
+        self,
+        col_name_or_expr: Union[str, ColumnExpression],
+        salting_partitions=None,
+        arrays_to_explode=None,
+    ):
+        super().__init__(
+            salting_partitions=salting_partitions, arrays_to_explode=arrays_to_explode
+        )
+        self.col_expression = ColumnExpression.instantiate_if_str(col_name_or_expr)
+
+    def create_sql(self, sql_dialect: SplinkDialect) -> str:
+        self.col_expression.sql_dialect = sql_dialect
+        col = self.col_expression
+        return f"{col.l_name} = {col.r_name}"
+
+
+class CustomRule(BlockingRuleCreator):
+    def __init__(
+        self,
+        sql_condition: str,
+        base_dialect_str: str = None,
+        salting_partitions=None,
+        arrays_to_explode=None,
+    ):
+        super().__init__(
+            salting_partitions=salting_partitions, arrays_to_explode=arrays_to_explode
+        )
+        self.sql_condition = sql_condition
+
+        self.base_dialect_str = base_dialect_str
+
+    def create_sql(self, sql_dialect: SplinkDialect) -> str:
```

Comment:
I've copied this across rather than attempt to re-use the code from the [CustomLevel](https://github.com/moj-analytical-services/splink/blob/72afcf009a30aa2eacb58bb6694b8a7adc2ea348/splink/comparison_level_library.py#L138)

That was deliberate, to reduce indirection, but i don't feel particularly strongly about this